packages/xstate-inspect/src/browser.ts

```diff
@@ -41,9 +41,11 @@ export function createDevTools(): XStateDevInterface {
       serviceMap.set(service.sessionId, service);
       serviceListeners.forEach((listener) => listener(service));
 
-      service.onStop(() => {
-        services.delete(service);
-        serviceMap.delete(service.sessionId);
+      service.subscribe({
+        complete() {
+          services.delete(service);
+          serviceMap.delete(service.sessionId);
+        }
```

Comment:
Likely the same thing should happen for the `error` notification as well - as both of those are the "final" ones. I'm OK with not dealing with this here - but we could add a comment here (or somewhere) to handle this for all of those subscriptions in `@xstate/inspect`